Swiftz/DictionaryExt.swift

```diff
@@ -6,14 +6,24 @@
 //  Copyright (c) 2014 Maxwell Swadling. All rights reserved.
 //
 
+extension Array {
+	/// Creates a dictionary from the array given a transformer function that returns 
+	/// a (key, value) pair for the dictionary, if nil is returned then the value is
+	/// not added to the dictionary.
+	public func mapKeyValues<K, V>(f : Element -> (K, V)?) -> [K: V] {
```

Comment:
I haven't heard of label, but that could make sense in that you're applying a label (a key) to each value.